azure-iot-device/azure/iot/device/common/transport/mqtt/pipeline_ops.py

```diff
@@ -0,0 +1,342 @@
+    This operation is in the group of "IotHub" operations becauase it is very specific to the IotHub client
+    """
+
+    def __init__(self, message=None, callback=None):
+        """
+        Constructor to instantiate a `IotHub_SendTelemetry` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.message = message
+        self.needs_connection = True
+
+
+class IotHub_SendOutputEvent(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to send an output message to an EdgeHub server.
+
+    This operation is in the group of "IotHub" operations becauase it is very specific to the IotHub client
+    """
+
+    def __init__(self, message=None, callback=None):
+        """
+        Constructor to instantiate a `IotHub_SendOutputEvent` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.message = message
+        self.needs_connection = True
+
+
+class Mqtt_SetConnectionArgs(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to connect to a server using the Mqtt protocol.
+
+    This operation is in the group of "Mqtt" operations becauase its attributes are very specific to the Mqtt protocol.
+    """
+
+    def __init__(
+        self,
+        client_id=None,
+        hostname=None,
+        username=None,
+        device_id=None,
+        module_id=None,
+        ca_cert=None,
+        callback=None,
+    ):
+        """
+        Constructor to instantiate a `Mqtt_SetConnectionArgs` object.
+
+        :param callback Function: The function that gets called when this operation is complete or has failed.  The `callback` function must accept A PipelineOperation object which indicates the specific call which has completed or failed.
+        """
+        PipelineOperation.__init__(self, callback=callback)
+        self.client_id = client_id
+        self.hostname = hostname
+        self.username = username
+        self.device_id = device_id
+        self.module_id = module_id
+        self.ca_cert = ca_cert
+
+    pass
+
+
+class Mqtt_Publish(PipelineOperation):
+    """
+    A PipelineOperation object which contains arguments used to publish a specific payload on a specific topic using the Mqtt protocol.
+
+    This operation is in the group of "Mqtt" operations becauase its attributes are very specific to the Mqtt protocol.
+    """
+
+    def __init__(self, topic=None, payload=None, qos=1, callback=None):
```

Comment:
>qos=1 [](start = 49, length = 5)

Why let this be settable? Do we ever want to have a specific Publish with a unique QoS? Why not just hardcode the Publish QoS as 1?

Reply:
We should.  we carry this around and don't need to.

---
In reply to: [279685064](https://github.com/Azure/azure-iot-sdk-python-preview/pull/87#discussion_r279685064) [](ancestors = 279685064)